src/GovUk.Education.ExploreEducationStatistics.Data.Api/Services/PermalinkService.cs

```diff
@@ -110,44 +106,62 @@ private async Task<PermalinkViewModel> BuildViewModel(Permalink permalink)
         {
             var viewModel = _mapper.Map<PermalinkViewModel>(permalink);
 
-            viewModel.Invalidated = !await IsValid(permalink.Query.SubjectId);
+            viewModel.Status = await GetPermalinkStatus(permalink.Query.SubjectId);
 
             return viewModel;
         }
 
-        private async Task<bool> IsValid(Guid subjectId)
+        private async Task<PermalinkStatus> GetPermalinkStatus(Guid subjectId)
         {
             var subject = await _subjectRepository.Get(subjectId);
             if (subject == null)
             {
-                return false;
+                return PermalinkStatus.SubjectRemoved;
             }
 
-            var releaseSubject = _statisticsDbContext.ReleaseSubject
-                .Include(rs => rs.Release)
-                .Where(rs => rs.SubjectId == subjectId)
-                .ToList()
-                .SingleOrDefault(rs =>
-                    _releaseRepository.IsLatestVersionOfRelease(rs.Release.PublicationId, rs.Release.Id));
-            if (releaseSubject == null)
+            var releasesWithSubject = _contentDbContext.ReleaseFiles
+                .Include(rf => rf.File)
+                .Include(rf => rf.Release)
+                .ThenInclude(r => r.Publication)
+                .ThenInclude(p => p.Releases)
+                .Where(rf =>
+                    rf.File.SubjectId == subjectId
+                    && rf.File.Type == FileType.Data
+                    && rf.Release.Published.HasValue && DateTime.UtcNow >= rf.Release.Published.Value)
+                .Select(rf => rf.Release)
+                .ToList();
+            if (releasesWithSubject.Count == 0)
             {
-                return false;
+                throw new Exception("The subject should be attached to at least one live release");
```

Comment:
This might change depending on whether you agree with my other comment and have to rework this or not, but I don't think we should be throwing any exceptions here. We have a Permalink and regardless of the state of any of the data in our database we should find a way of showing that table.